src/internal_modules/roc_core/memory_limiter.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/memory_limiter.h
+//! @brief Memory limiter.
+
+#ifndef ROC_CORE_MEMORY_LIMITER_H_
+#define ROC_CORE_MEMORY_LIMITER_H_
+
+#include "roc_core/atomic.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/stddefs.h"
+
+namespace roc {
+namespace core {
+
+//! Memory limiter.
+class MemoryLimiter : public NonCopyable<> {
+public:
+    //! Initialize memory limiter.
+    explicit MemoryLimiter(const char* name, size_t max_bytes);
+
+    //! Destroy memory limiter.
+    ~MemoryLimiter();
+
+    //! Acquire memory.
+    bool acquire(size_t num_bytes);
+
+    //! Release memory.
+    void release(size_t num_bytes);
+
+    //! Get number of bytes currently acquired.
+    size_t acquired();
```

Comment:
I guess we can make it const.

Reply:
Also usually we name such method "num_acquired" or "acquired_count" or something like this, to make it clear that it's not boolean.